pkg/providers/property/data_akamai_search.go

```diff
@@ -0,0 +1,115 @@
+package property
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/akamai/AkamaiOPEN-edgegrid-golang/v2/pkg/papi"
+	"github.com/akamai/terraform-provider-akamai/v2/pkg/akamai"
+	"github.com/akamai/terraform-provider-akamai/v2/pkg/tools"
+)
+
+func dataSourceSearch() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceSearchRead,
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Key must have one of three values: \"edgeHostname\", \"hostname\" or \"propertyName\"",
+				ValidateDiagFunc: tools.IsNotBlank,
+			},
+			"value": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Value of Search",
+				ValidateDiagFunc: tools.IsNotBlank,
+			},
+			"properties": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of properties",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"account_id":        {Type: schema.TypeString, Computed: true},
+						"asset_id":          {Type: schema.TypeString, Computed: true},
+						"contract_id":       {Type: schema.TypeString, Computed: true},
+						"group_id":          {Type: schema.TypeString, Computed: true},
+						"property_id":       {Type: schema.TypeString, Computed: true},
+						"property_version":  {Type: schema.TypeInt, Computed: true},
+						"property_name":     {Type: schema.TypeString, Computed: true},
+						"edge_hostname":     {Type: schema.TypeString, Computed: true},
+						"hostname":          {Type: schema.TypeString, Computed: true},
+						"production_status": {Type: schema.TypeString, Computed: true},
+						"staging_status":    {Type: schema.TypeString, Computed: true},
+						"updated_by_user":   {Type: schema.TypeString, Computed: true},
+						"updated_date":      {Type: schema.TypeString, Computed: true},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceSearchRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	meta := akamai.Meta(m)
+	client := inst.Client(meta)
+
+	log := meta.Log("PAPI", "dataSourceSearchRead")
+
+	log.Debug("Read Search")
+
+	var key, value string
+	var err error
+
+	if key, err = tools.GetStringValue("key", d); err != nil {
```

Comment:
```suggestion
	if key, err := tools.GetStringValue("key", d); err != nil {
```

Reply:
this will also change the scope ok key, right? If you want, I can either firstly run the function, then check the error response, or keep the var error definition before

Reply:
This is rather personal preference. I like to reduce the scope of variables, and if we can do that to the minimum (within the `if` block) even better.

This way we reduce the possibility of forgetting to handle errors: if we are reusing the `err` variable, compiler may not complain if in one of those cases we didn't check its value.

Of course in this case you did not forget any of that; but we tend to be quite tight with code reviews in our team :-D

Reply:
Hi Fuochi,

I am sorry, busy day as you may have seen in the repository.

I guess that we can live with this. I am looking at this code in my local editor and overall it looks fine.

I will resolve this and all related conversations in the PR. @Slonimskaia may add some new comments if she still has concerns.